### DISCUSSION TITLE
add lock/confirmation on Cover

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#8947] [Dashboard]
Please consider as title, since the slider can be used for home entering garage doors too.... Id really love to be able to guard that with a secondary click on a confirmation:

<img width="426" alt="Schermafbeelding 2021-04-19 om 10 56 35" src="https://user-images.githubusercontent.com/33354141/115210441-dde76980-a0fe-11eb-820e-ba49f5a4e5bd.png">

right now this is a bit too prone to accidental opening
request would be both for the entities (top) and entity card (bottom) on a cover entity.

thanks for considering
(I've FR'd the slider-entity-row repo with the exact same request)

**Comment by iantrich** [comment]
History: https://github.com/home-assistant/frontend/pull/3994

**Reply by Mariusthvdb** [reply]
yes, as I have done right now using your restriction card.

my Fr was about a core lock on Cover in this case, but could/should maybe even be broader, on more entities.

as Balloob suggested here https://github.com/home-assistant/frontend/pull/3994#issuecomment-541389996 this is probably now settled enough to adopt in core?

Really feel a mature Home Automation platform should offer its own lock option, especially on those Home security sensitive devices as doors and windows.

**Reply by iantrich** [reply]
The main issue I see is the sense of false security as it is easy bypassed.

**Reply by Mariusthvdb** [reply]
Yes, I see that. Not 100% secure.

Though, one would really need to know his/her way in the allowed interface a correct design displays to unauthorized people. 
In my use case it also there for preventing accidental opening, which happened more easily than wished for.

Maybe compare with the Google secure devices? That's not unbreakable either, and still a very nice first hurdle.

And, maybe it can be improved upon? I wasn't implying your exact code needs to be integrated into core. (Please don't take offense).
I was hoping for a core implementation of lock, which of course needs to be as secure as possible given the options Core offers...

**Reply by iantrich** [reply]
I think the only way to make such things truly secure is to make it part of the backend integrations. Doing this within Lovelace will not be secure.

**Reply by Mariusthvdb** [reply]
well, yes, probably. 

That doesn't take away from the fact though we fail a core frontend 'lock' now, not to make things secure, as in 100% burglar proof, but as in 'prevent from accidental operation' at least. And, make it more difficult for non authorized bystanders  along the way. And that was the purpose of my FR, not creating a 100% secure backend.

A visual (opening/closing) lock, with a confirmation, like [already available](https://www.home-assistant.io/lovelace/actions/#options-for-confirmation) on the core  button row element would be very welcome in core. 

Add in the pin option, and it would be even better. Could this Pin option be added to the current action/confirmation options? Would that need a separate FR? I'd be happy to create that.

I anything, please allow for the confirmation on action for these covers then?
```
      - entity: cover.garage_door
        secondary_info: last-changed
        tap_action:
          confirmation:
            text: Are you sure to operate the Garage door?
```

Btw, talking about secure, one could have the same discussion in the core backend [alarm_panel](https://www.home-assistant.io/integrations/manual). in itself this is not secure at all, and its all up to the end user to make it so.

**Reply by spacegaier** [reply]
Similar topic: https://github.com/home-assistant/frontend/discussions/8905. I want to implement the confirmation for toggles.

**Reply by Mariusthvdb** [reply]
Thanks, let me add there, and link to this, for my request on the lock/confirmation on covers. seems a special category (both in frontend, and because of the nature of the device)
